Override inspect in Combined, Env & Encrypted Configurations

siaw23 opened the pull request:
Previously, inspecting CombinedConfiguration and EnvConfiguration,
would expose all sensitive data including ENV 
variables, credentials and internal object state. This poses a security 
risk when objects are displayed in console, logs, or error messages.

Before:
```ruby
    Rails.app.creds
    #<ActiveSupport::CombinedConfiguration:0x0000000122b45870
     @configurations=[
       #<ActiveSupport::EnvConfiguration:0x0000000122b465b8
        @envs={"SOME_SECRET" => "secret_value", "DATABASE_URL" => "postgresql://user:pass@host/db", ...}>,
       #<ActiveSupport::EncryptedConfiguration:0x00000000007af0>]>
```
All sensitive values were exposed in the output, making it easy to 
accidentally leak secrets in logs or error reports.

After:
```ruby
    Rails.app.creds
    #<ActiveSupport::CombinedConfiguration:0x00000000006490 keys=[:rails_env, :some_secret, ..., :secret_key_base, :stripe_webhook_secret]>

    Rails.app.credentials
    #<ActiveSupport::EncryptedConfiguration:0x00000000008cd0 keys=[:secret_key_base, :stripe_webhook_secret]>

    Rails.app.envs
    #<ActiveSupport::EnvConfiguration:0x0000000000a6a0 keys=[:rails_env, :some_secret, :path, :home, ...]>
```
Now only key names are shown, displayed as symbols (in EncryptedConfiguration too) to match how they're 
accessed in code.

cc: @dhh 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

siaw23 commented:
@dhh I made some changes so for my local dummy app we now have something like: 

```
kreds(dev):001> Rails.app.creds
=> #<ActiveSupport::CombinedConfiguration:0x000000000060a0 keys=["RBENV_VERSION", "NVM_INC", "TERM_PROGRAM", "NVM_CD_FLAGS", "TERM", "SHELL", "HOMEBREW_REPOSITORY", "TMPDIR", "TERM_PROGRAM_VERSION", "TERM_SESSION_ID", "ZSH", "NVM_DIR", "USER", "LS_COLORS", "COMMAND_MODE", "RBENV_ROOT", "SSH_AUTH_SOCK", "__CF_USER_TEXT_ENCODING", "TERM_FEATURES", "PAGER", "RBENV_HOOK_PATH", "LSCOLORS", "TERMINFO_DIRS", "PATH", "LaunchInstanceID", "RBENV_ORIG_PATH", "__CFBundleIdentifier", "PWD", "LANG", "ITERM_PROFILE", "XPC_FLAGS", "RBENV_SHELL", "XPC_SERVICE_NAME", "SHLVL", "HOME", "COLORFGBG", "LC_TERMINAL_VERSION", "HOMEBREW_PREFIX", "RBENV_DIR", "ITERM_SESSION_ID", "LESS", "LOGNAME", "NVM_BIN", "INFOPATH", "HOMEBREW_CELLAR", "LC_TERMINAL", "RUBYLIB", "SECURITYSESSIONID", "COLORTERM", "BUNDLE_GEMFILE", "BUNDLER_ORIG_BUNDLE_BIN_PATH", "BUNDLER_ORIG_BUNDLE_GEMFILE", "BUNDLER_ORIG_BUNDLER_VERSION", "BUNDLER_ORIG_BUNDLER_SETUP", "BUNDLER_ORIG_GEM_HOME", "BUNDLER_ORIG_GEM_PATH", "BUNDLER_ORIG_MANPATH", "BUNDLER_ORIG_PATH", "BUNDLER_ORIG_RB_USER_INSTALL", "BUNDLER_ORIG_RUBYLIB", "BUNDLER_ORIG_RUBYOPT", "GEM_HOME", "BUNDLE_BIN_PATH", "BUNDLER_VERSION", "BUNDLER_SETUP", "RUBYOPT", "RAILS_ENV", "SOME_SECRET", :secret_key_base, :stripe_webhook_secret]>
kreds(dev):002> Rails.app.credentials
=> #<ActiveSupport::EncryptedConfiguration:0x000000000088e0 keys=[:secret_key_base, :stripe_webhook_secret]>
kreds(dev):003> Rails.app.envs
=> #<ActiveSupport::EnvConfiguration:0x0000000000a2b0 keys=["RBENV_VERSION", "NVM_INC", "TERM_PROGRAM", "NVM_CD_FLAGS", "TERM", "SHELL", "HOMEBREW_REPOSITORY", "TMPDIR", "TERM_PROGRAM_VERSION", "TERM_SESSION_ID", "ZSH", "NVM_DIR", "USER", "LS_COLORS", "COMMAND_MODE", "RBENV_ROOT", "SSH_AUTH_SOCK", "__CF_USER_TEXT_ENCODING", "TERM_FEATURES", "PAGER", "RBENV_HOOK_PATH", "LSCOLORS", "TERMINFO_DIRS", "PATH", "LaunchInstanceID", "RBENV_ORIG_PATH", "__CFBundleIdentifier", "PWD", "LANG", "ITERM_PROFILE", "XPC_FLAGS", "RBENV_SHELL", "XPC_SERVICE_NAME", "SHLVL", "HOME", "COLORFGBG", "LC_TERMINAL_VERSION", "HOMEBREW_PREFIX", "RBENV_DIR", "ITERM_SESSION_ID", "LESS", "LOGNAME", "NVM_BIN", "INFOPATH", "HOMEBREW_CELLAR", "LC_TERMINAL", "RUBYLIB", "SECURITYSESSIONID", "COLORTERM", "BUNDLE_GEMFILE", "BUNDLER_ORIG_BUNDLE_BIN_PATH", "BUNDLER_ORIG_BUNDLE_GEMFILE", "BUNDLER_ORIG_BUNDLER_VERSION", "BUNDLER_ORIG_BUNDLER_SETUP", "BUNDLER_ORIG_GEM_HOME", "BUNDLER_ORIG_GEM_PATH", "BUNDLER_ORIG_MANPATH", "BUNDLER_ORIG_PATH", "BUNDLER_ORIG_RB_USER_INSTALL", "BUNDLER_ORIG_RUBYLIB", "BUNDLER_ORIG_RUBYOPT", "GEM_HOME", "BUNDLE_BIN_PATH", "BUNDLER_VERSION", "BUNDLER_SETUP", "RUBYOPT", "RAILS_ENV", "SOME_SECRET"]>
```

dhh commented on activesupport/lib/active_support/encrypted_configuration.rb:
```diff
@@ -130,7 +130,14 @@ def config
     def inspect # :nodoc:
+      "#<#{self.class.name}:#{'%#016x' % (object_id << 1)} keys=#{keys.inspect}>"
+    rescue
       "#<#{self.class.name}:#{'%#016x' % (object_id << 1)}>"
```
What are we rescuing?

dhh commented on activesupport/CHANGELOG.md:
````diff
@@ -1,22 +1,30 @@
     ```ruby
     Rails.app.creds
-    => #<ActiveSupport::CombinedConfiguration:0x0000000122b45870>
+    #<ActiveSupport::CombinedConfiguration:0x00000000006490 keys=["RAILS_ENV", "SOME_SECRET", ..., :secret_key_base, :stripe_webhook_secret]>
````
We should be converting all these keys to symbols, since that's how you're going to access them.

dhh commented on activesupport/CHANGELOG.md:
````diff
@@ -1,3 +1,34 @@
+    #<ActiveSupport::EnvConfiguration:0x0000000000a6a0 keys=[:rails_env, :some_secret, :path, :home, ...]>
+    ```
+
+    *Emmanuel Hayford*
````
Since this is an unreleased feature, we don't need a CHANGELOG entry. But we can amend the other entries to note that these new configuration objects are inspect safe.

siaw23 committed:
Override inspect in Combined, Env & Encrypted Configurations

Previously, inspecting CombinedConfiguration and EnvConfiguration,
would expose all sensitive data including ENV
variables, credentials and internal object state. This poses a security
risk when objects are displayed in console, logs, or error messages.

Before:

    Rails.app.creds
    #<ActiveSupport::CombinedConfiguration:0x0000000122b45870
     @configurations=[
       #<ActiveSupport::EnvConfiguration:0x0000000122b465b8
        @envs={"SOME_SECRET" => "secret_value", "DATABASE_URL" => "postgresql://user:pass@host/db", ...}>,
       #<ActiveSupport::EncryptedConfiguration:0x00000000007af0>]>

All sensitive values were exposed in the output, making it easy to
accidentally leak secrets in logs or error reports.

After:

    Rails.app.creds
    #<ActiveSupport::CombinedConfiguration:0x00000000006490 keys=[:rails_env, :some_secret, ..., :secret_key_base, :stripe_webhook_secret]>

    Rails.app.credentials
    #<ActiveSupport::EncryptedConfiguration:0x00000000008cd0 keys=[:secret_key_base, :stripe_webhook_secret]>

    Rails.app.envs
    #<ActiveSupport::EnvConfiguration:0x0000000000a6a0 keys=[:rails_env, :some_secret, :path, :home, ...]>

Now only key names are shown, displayed as symbols (in EncryptedConfiguration too) to match how they're
accessed in code.
